react/ImageList.tsx

```diff
@@ -0,0 +1,116 @@
+import React, { createContext, useContext } from 'react'
+import { defineMessages } from 'react-intl'
+import { useDevice } from 'vtex.device-detector'
+
+import Image from './Image'
+
+interface Props {
+  images: {
+    link?: {
+      url: string
+      noFollow: boolean
+      openNewTab: boolean
+      title: string
+    }
+    image: string
+    mobileImage: string
+    description: string
+  }[]
+  height: number
+}
+
+const ImageListStateContext = createContext<JSX.Element[] | undefined>(
+  undefined
+)
+
+const ImageList: StorefrontFunctionComponent<Props> = ({
+  images,
+  height = 420,
+  children,
+}) => {
+  const { isMobile } = useDevice()
+  const imageListContent = images.map(
+    ({ link, image, mobileImage, description }, idx) => (
+      <Image
+        key={idx}
+        src={isMobile && mobileImage ? mobileImage : image}
+        alt={description}
+        link={link}
+        maxHeight={height}
+        width="100%"
+      />
+    )
+  )
+
+  return (
+    <ImageListStateContext.Provider value={imageListContent}>
+      {children}
+    </ImageListStateContext.Provider>
+  )
+}
+
+export function useImageListState() {
+  const context = useContext(ImageListStateContext)
+  return context
```

Comment:
```suggestion
  return useContext(ImageListStateContext)
```

Reply:
Nice! But I think I'll add a conditional check to account for using this hook when there is no `ImageListStateContext` directly above the component calling this. :)